src/commonMain/kotlin/app/cash/turbine/FlowTurbine.kt

```diff
@@ -303,3 +334,24 @@ private class AssertionError(
   message: String,
   override val cause: Throwable?
 ) : kotlin.AssertionError(message)
+
+/**
+ * This class helps us to distinguish a not received item from a received item with a null
+ * value. A nicer approach could be to use a `lateinit` field and invoke the `isInitialized`
+ * field, but unfortunately, this cannot happen in a function scope.
+ */
+private class ItemWrapper<T> private constructor(private val event: Event.Item<T>? = null) {
```

Comment:
Hmm I don't understand what this class offers over using the `ChannelResult` from `tryReceive()`. Doesn't it already allow disambiguating no value vs. null value?

Reply:
You are right about that! We can achieve the same result without using an extra class here (aka `ItemWrapper`) 😄